frame/balances/src/lib.rs

```diff
@@ -160,106 +160,240 @@ use sp_std::prelude::*;
 pub use weights::WeightInfo;
 
-pub trait Subtrait<I: Instance = DefaultInstance>: frame_system::Config {
-	/// The balance of an account.
-	type Balance: Parameter + Member + AtLeast32BitUnsigned + Codec + Default + Copy +
-		MaybeSerializeDeserialize + Debug;
+pub use pallet::*;
 
-	/// The minimum amount required to keep an account open.
-	type ExistentialDeposit: Get<Self::Balance>;
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+	use super::*;
 
-	/// The means of storing the balances of an account.
-	type AccountStore: StoredMap<Self::AccountId, AccountData<Self::Balance>>;
+	#[pallet::config]
+	pub trait Config<I: 'static = ()>: frame_system::Config {
+		/// The balance of an account.
+		type Balance: Parameter + Member + AtLeast32BitUnsigned + Codec + Default + Copy +
+			MaybeSerializeDeserialize + Debug;
 
-	/// Weight information for the extrinsics in this pallet.
-	type WeightInfo: WeightInfo;
+		/// Handler for the unbalanced reduction when removing a dust account.
+		type DustRemoval: OnUnbalanced<NegativeImbalance<Self, I>>;
 
-	/// The maximum number of locks that should exist on an account.
-	/// Not strictly enforced, but used for weight estimation.
-	type MaxLocks: Get<u32>;
-}
+		/// The overarching event type.
+		type Event: From<Event<Self, I>> + IsType<<Self as frame_system::Config>::Event>;
 
-pub trait Config<I: Instance = DefaultInstance>: frame_system::Config {
-	/// The balance of an account.
-	type Balance: Parameter + Member + AtLeast32BitUnsigned + Codec + Default + Copy +
-		MaybeSerializeDeserialize + Debug;
+		/// The minimum amount required to keep an account open.
+		#[pallet::constant]
+		type ExistentialDeposit: Get<Self::Balance>;
 
-	/// Handler for the unbalanced reduction when removing a dust account.
-	type DustRemoval: OnUnbalanced<NegativeImbalance<Self, I>>;
+		/// The means of storing the balances of an account.
+		type AccountStore: StoredMap<Self::AccountId, AccountData<Self::Balance>>;
 
-	/// The overarching event type.
-	type Event: From<Event<Self, I>> + Into<<Self as frame_system::Config>::Event>;
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
 
-	/// The minimum amount required to keep an account open.
-	type ExistentialDeposit: Get<Self::Balance>;
+		/// The maximum number of locks that should exist on an account.
+		/// Not strictly enforced, but used for weight estimation.
+		type MaxLocks: Get<u32>;
+	}
 
-	/// The means of storing the balances of an account.
-	type AccountStore: StoredMap<Self::AccountId, AccountData<Self::Balance>>;
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T, I=()>(PhantomData<(T, I)>);
 
-	/// Weight information for extrinsics in this pallet.
-	type WeightInfo: WeightInfo;
+	#[pallet::hooks]
+	impl<T: Config<I>, I: 'static> Hooks<BlockNumberFor<T>> for Pallet<T, I> {
+	}
```

Comment:
Is this empty block necessary or could we just remove it?

Reply:
It's mandatory https://crates.parity.io/frame_support/attr.pallet.html#hooks-pallethooks-mandatory